scipy/stats/_hypotests.py

```diff
@@ -1393,3 +1396,692 @@ def cramervonmises_2samp(x, y, method='auto'):
+    For paired sample statistics, two null hypothesis can be tested:
+    that the data are paired at random or that the data are assigned to samples
+    at random.
+
+    Parameters
+    ----------
+    data : iterable of array-like
+        Contains the samples, each of which is an array of observations.
+        Dimensions of sample arrays must be compatible for broadcasting except
+        along `axis`.
+    statistic : callable
+        Statistic for which the p-value of the hypothesis test is to be
+        calculated. `statistic` must be a callable that accepts samples
+        as separate arguments (e.g. ``statistic(*data)``) and returns the
+        resulting statistic.
+        If `vectorized` is set ``True``, `statistic` must also accept a keyword
+        argument `axis` and be vectorized to compute the statistic along the
+        provided `axis` of the sample arrays.
+    permutation_type : {'both', 'samples', 'pairings'}, optional
+        The type of permutations to be performed, in accordance with the
+        null hypothesis. The first two permutation types are for paired sample
+        statistics, in which all samples contain the same number of
+        observations and observations with corresponding indices along `axis`
+        are considered to be paired; the third is for independent sample
+        statistics.
+
+        - ``'samples'`` : observations are assigned to different samples
+          but remain paired with the same observations from other samples.
+          This permutation type is appropriate for paired sample hypothesis
+          tests such as the Wilcoxon signed-rank test and the paired t-test.
+        - ``'pairings'`` : observations are paired with different observations,
+          but they remain within the same sample. This permutation type is
+          appropriate for association/correlation tests with statistics such
+          as Spearman's :math:`\rho`, Kendall's :math:`\tau`, and Pearson's
+          :math:`r`.
+        - ``'both'`` (default) : observations are assigned to different samples
+          without preserving pairs. Samples may contain different numbers of
+          observations. This permutation type is appropriate for
+          independent sample hypothesis tests such as the Mann-Whitney
+          :math:`U` test and the independent sample t-test.
+
+          Please see the Notes section below for more detailed descriptions
+          of the permutation types.
+
+    vectorized : bool, default: ``False``
+        By default, `statistic` is assumed to calculate the statistic only for
+        1D arrays contained in `data`. If `vectorized` is set ``True``,
+        `statistic` must also accept a keyword argument `axis` and be
+        vectorized to compute the statistic along the provided `axis` of the ND
+        arrays in `data`. Use of a vectorized statistic can reduce computation
+        time.
+    permutations : int, default: ``np.inf``
```

Comment:
Something to consider before merging is renaming the `permutations` parameter to `n_resamples` to keep this consistent with `bootstrap` (and gh-14576). It would be different from the name in `ttest_ind`, but I think it's more important to keep the names consistent among these "resampling methods". The plan (after this PR merges) would be to rename `_bootstrap.py` to `_resampling.py` and move this over there.

Reply:
any suggestion how to proceed? somehow `permutations` or `n_permutations` seems better than `n_resamples` since I would not directly associate the latter with permutations

Reply:
Well, permutation testing is one of several ["resampling methods"](https://en.wikipedia.org/wiki/Resampling_(statistics)#Permutation_tests) in statistics, so I think `n_resamples` is definitely appropriate, and it would be more consistent with bootstrap.  I'd suggest we either leave it as-is or change it to `n_resamples`. Your call!

Reply:
Ok, true, should we change it to `n_resamples` then?

Reply:
Yes, I'll do that and change "both" to "independent" today!

Reply:
Done.